File: lib/to_mdx/parsers/open_document/office/styles.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Office
      class Styles < Base
        def initialize(element = nil, context = {})
          super(element, context)
        end

        def to_css
          children.map(&:to_css).reduce({}, :merge)
        end

        register_for "office:styles"
      end
    end
  end
end

File: src/parsers/open-document/office/styles.ts
import { BaseElement } from "../../base_element.ts";

export class Styles extends BaseElement {
  toMdx(): string {
    return "";
  }

  toString(): string {
    return "";
  }

  toCss(): Record<string, unknown> {
    return this.children
      .map((child) => {
        if ("toCss" in child && typeof child.toCss === "function") {
          return (child as unknown as { toCss: () => Record<string, unknown> }).toCss();
        }
        return {};
      })
      .reduce((acc, styles) => ({ ...acc, ...styles }), {});
  }

  static {
    BaseElement.registerFor.call(this, "office:styles");
  }
}
